owl/SOMA-OBJ.owl

```diff
@@ -2431,6 +2493,20 @@ Resources are often consumed by their participation in an event, but this need n
 
     <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#ShapeRegion">
         <rdfs:subClassOf rdf:resource="http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#Region"/>
+        <rdfs:subClassOf>
+            <owl:Restriction>
+                <owl:onProperty rdf:resource="http://www.ease-crc.org/ont/SOMA-OBJ.owl#hasReferencePose"/>
+                <owl:qualifiedCardinality rdf:datatype="http://www.w3.org/2001/XMLSchema#nonNegativeInteger">1</owl:qualifiedCardinality>
+                <owl:onDataRange rdf:resource="http://www.w3.org/2001/XMLSchema#string"/>
+            </owl:Restriction>
+        </rdfs:subClassOf>
+        <rdfs:subClassOf>
+            <owl:Restriction>
+                <owl:onProperty rdf:resource="http://www.ease-crc.org/ont/SOMA-OBJ.owl#hasShapeScale"/>
```

Comment:
In KnowRob, the scale is only a property of mesh shapes.
Do we need scale for anything other then the mesh shapes?

Reply:
In principle, it is possible to apply scaling to any geometric primitive. Of course, this can result in the shape being no longer the initial primitive (e.g. a sphere becomes an ellipsoid). As far as I can tell, for example RViz uses scale.x, scale.y, scale.z parameters in a marker message even for some of the basic shapes such as the cylinder.

OTOH, we don't need to do this. Cylinders can be specified by radius, height; spheres are not ellipsoids. So to keep things simple we can restrict scaling to meshes.